edk2toollib/windows/locate_tools.py

```diff
@@ -135,11 +137,16 @@ def FindWithVsWhere(products="*"):
 # keys: enumerable list with names of env variables to collect after bat run
 # arch: arch to run.  amd64, x86, ??
 # returns a dictionary of the interesting environment variables
-def QueryVcVariables(keys, arch="amd64"):
+def QueryVcVariables(keys: dict, arch: str = None, product: str = None):
     """Launch vcvarsall.bat and read the settings from its environment"""
+    if product is None:
+        product = "*"
+    if arch is None:
+        # TODO: look up host architecture?
+        arch = "amd64"
     interesting = set(keys)
     result = {}
-    ret, vs_path = FindWithVsWhere()
+    ret, vs_path = FindWithVsWhere(product)
     if ret != 0:
```

Comment:
You need to check if vs_path is not none. It looks like we don't return a non zero ret in all cases

Reply:
done